### PR TITLE
enhance the sdk of cce addon to support listing addon instances

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

enhance the sdk of cce addon to support listing addon instances

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: openstack/cce/v3/addons/urls.go [hunk 1]
```diff
@@ -18,6 +18,10 @@ func resourceURL(client *golangsdk.ServiceClient, id, cluster_id string) string
 	return CCEServiceURL(client, cluster_id, rootPath, id+"?cluster_id="+cluster_id)
 }
 
+func resourceListURL(client *golangsdk.ServiceClient, cluster_id string) string {
+	return CCEServiceURL(client, cluster_id, rootPath+"?cluster_id="+cluster_id)
+}
+
 func CCEServiceURL(client *golangsdk.ServiceClient, cluster_id string, parts ...string) string {
 	rbUrl := "https://" + cluster_id + "." + client.ResourceBaseURL()[8:]
```
Comment (ShiChangkuo) [inline openstack/cce/v3/addons/urls.go:26]: this line is hard to understand, suggest to use `net/url` package to generate it.

File: openstack/cce/v3/addons/requests.go [hunk 2]
```diff
@@ -1,13 +1,94 @@
...
+	AddonTemplateName string `json:"addonTemplateName"`
+	Uid               string `json:"uid"`
+	Version           string `json:"version"`
+	Status            string `json:"status"`
+}
+
+// List returns collection of addons.
+func List(client *golangsdk.ServiceClient, clusterID string, opts ListOpts) ([]Addon, error) {
+	var r ListResult
+	_, r.Err = client.Get(resourceListURL(client, clusterID), &r.Body, &golangsdk.RequestOpts{
```
Comment (ShiChangkuo) [inline openstack/cce/v3/addons/requests.go:24]: dose the RequestOpts is essential？

File: openstack/cce/v3/addons/requests.go [hunk 3]
```diff
@@ -1,13 +1,94 @@
...
+				if sVal := GetStructNestedField(&addon, key, value.Driller); !(sVal == value.Value) {
+					matched = false
+				}
+			}
+			if matched {
+				refinedAddons = append(refinedAddons, addon)
+			}
+		}
+	} else {
+		refinedAddons = addons
```
Comment (ShiChangkuo) [inline openstack/cce/v3/addons/requests.go:72]: looks this line is unnecessary.

File: openstack/cce/v3/addons/requests.go [hunk 4]
```diff
@@ -1,13 +1,94 @@
...
+	if opts.Status != "" {
+		m["Status"] = FilterStruct{Value: opts.Status, Driller: []string{"Status"}}
+	}
+
+	if len(m) > 0 && len(addons) > 0 {
+		for _, addon := range addons {
+			matched = true
+
+			for key, value := range m {
+				if sVal := GetStructNestedField(&addon, key, value.Driller); !(sVal == value.Value) {
```
Comment (ShiChangkuo) [inline openstack/cce/v3/addons/requests.go:63]: s/`!(sVal == value.Value)`/`sVal != value.Value`/ ?

File: openstack/cce/v3/addons/requests.go [hunk 5]
```diff
@@ -1,13 +1,94 @@
...
+					matched = false
```
Comment (ShiChangkuo) [inline openstack/cce/v3/addons/requests.go:64]: missing break?

**Review by ShiChangkuo — CHANGES_REQUESTED** [review]

File: openstack/cce/v3/addons/requests.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply openstack/cce/v3/addons/requests.go:63]: fixed

File: openstack/cce/v3/addons/requests.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply openstack/cce/v3/addons/requests.go:64]: fixed

File: openstack/cce/v3/addons/requests.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply openstack/cce/v3/addons/requests.go:24]: fixed

File: openstack/cce/v3/addons/urls.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply openstack/cce/v3/addons/urls.go:26]: fixed

**Review by ShiChangkuo — APPROVED** [review]

**Pull request merged**